hack/generators/config-graph/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"bytes"
+	"os"
+	"reflect"
+	"text/template"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/store"
+)
+
+// This code generates `fallback.NewConfigGraphFromCacheStores` function.
+
+const (
+	outputFile = "zz_generated_graph.go"
+
+	outputTemplate = `// Code generated by hack/generators/config-graph; DO NOT EDIT.
+package fallback
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dominikbraun/graph"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/store"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NewConfigGraphFromCacheStores(c store.CacheStores) (*ConfigGraph, error) {
+	g := graph.New[ObjectHash, client.Object](GetObjectHash, graph.Directed())
+
+	// Gather cache store fields for all kinds.
+	allStores := []cache.Store{
+		{{- range . }}
+		c.{{ . }},
+		{{- end }}
+	}
```

Comment:
Retroactive review of https://github.com/Kong/kubernetes-ingress-controller/pull/5993:

Does this actually need a template? Most of the code here isn't templated, and the part that is generates a simple slice:

```
 18   allStores := []cache.Store{                                                                                             
 19     c.IngressV1,                                                                                                          
 20     c.IngressClassV1,                                                                                                     
 21     c.Service,                                                                                                            
 22     c.Secret,                                                                                                             
 23     c.EndpointSlice,
...
```

Without any template control flow beyond the `range`. That slice is the same for both `cache-stores/main.go` and `config-graph/main.go`. Could we handle that instead with a `somepackage.GetAllStores()` function that returns the slice?

I might be missing some expected future work that will make the template more complicated--if that's the case some TODOs here would be good.

Reply:
That's a good point - generated `GetAllStores` should do the job for both cases.